lib/OpenQA/Worker.pm

```diff
@@ -603,11 +606,15 @@ sub is_qemu_running ($self) {
 }
 
 sub is_ovs_dbus_service_running ($self) {
-    eval { defined &Net::DBus::system or require Net::DBus };
-    return 0 if $@;
+    try { defined &Net::DBus::system or require Net::DBus }
+    catch ($e) { return 0 }
     my $bus = ($self->{_system_dbus} //= Net::DBus->system(nomainloop => 1));
-    my $service = eval { defined $bus->get_service('org.opensuse.os_autoinst.switch') };
-    return !$@ && $service;
+    my $err;
+    my $service = do {
+        try { defined $bus->get_service('org.opensuse.os_autoinst.switch') }
+        catch ($e) { $err = $e }
+    };
```

Comment:
this is not correct. The `do` block in error case would assign `$err` to `$service`

Instead you can directly return from within `try`

```suggestion
    try { return defined $bus->get_service('org.opensuse.os_autoinst.switch') }
    catch ($e) { return 0 }
```